ext/sodium/libsodium.c

```diff
@@ -0,0 +1,2682 @@
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include "php.h"
+#include "php_ini.h"
+#include "ext/standard/info.h"
+#include "php_libsodium.h"
+#include "zend_exceptions.h"
+
+#include <sodium.h>
+#include <stdint.h>
+
+#define ZSTR_TRUNCATE(zs, len) do { ZSTR_LEN(zs) = (len); } while(0)
```

Comment:
I don't think that it is a good idea to define a (global) Zend macro in an extension. This should rather be `PHP_SODIUM_ZSTR_TRUNCATE`, or moved to the other Zend macros.